.github/workflows/main.yml

```diff
@@ -107,3 +107,39 @@ jobs:
         echo "debian" $(cat /etc/debian_version)
         /usr/cyrus/libexec/master -V
         /usr/cyrus/sbin/cyr_buildinfo
+  valgrind-make-check:
+    runs-on: ubuntu-latest
+    container:
+        image: ghcr.io/cyrusimap/cyrus-docker:bookworm
+        options: --init
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: setup git safe directory
+      shell: bash
+      run: git config --global --add safe.directory /__w/cyrus-imapd/cyrus-imapd
+    - name: fetch upstream release tags
+      if: ${{ github.repository != 'cyrusimap/cyrus-imapd' }}
+      shell: bash
+      run: |
+        git remote add upstream https://github.com/cyrusimap/cyrus-imapd.git
+        # n.b. --no-tags does not mean "no tags", it means "no automatic tag
+        # following".  we're explicitly fetching the tags we want, we do not
+        # need every other tag that's reachable from them
+        git fetch --no-tags upstream 'refs/tags/cyrus-imapd-*:refs/tags/cyrus-imapd-*'
+    - name: configure and build
+      shell: bash
+      run: |
+        echo "building cyrus version" $(./tools/git-version.sh)
+        ./tools/build-with-cyruslibs.sh
+    - name: make check with valgrind
+      shell: bash
+      run: |
+        make VG=1 check
+    - name: report version information
```

Comment:
I don't think it matters, but it surprised me to have the version information at the end instead of before the tests. I guess this is probably the same order as the regular CI ("after build and check, before cassandane"), and the difference is that this one doesn't run cassandane

Reply:
Yeah... I've been mostly copying what we have and pinning it down. I tried to find a way to make this all reusable but it doesn't seem easy or obvious to modularize this stuff. /shrug